cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -186,4 +188,12 @@ ScheduleCreationSpec triggerOnProgramStatus(String application, ProgramType prog
    */
   ScheduleCreationSpec triggerOnProgramStatus(ProgramType programType, String program,
                                               ProgramStatus... programStatuses);
+
+  /**
+   * Creates a schedule with a given {@link ScheduleTrigger}
+   *
+   * @param trigger the trigger to be contained in the schedule
+   * @return a {@link ScheduleCreationSpec}
+   */
+  ScheduleCreationSpec withTrigger(ScheduleTrigger trigger);
```

Comment:
Following the same naming convention as other trigger methods, this one should be called `triggerOn`.